mucommander-core/src/main/java/com/mucommander/ui/text/EnhancedTextField.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * This file is part of muCommander, http://www.mucommander.com
+ *
+ * muCommander is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * muCommander is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.mucommander.ui.text;
+
+import javax.swing.*;
+import javax.swing.text.JTextComponent;
+import java.awt.event.FocusEvent;
+import java.awt.event.FocusListener;
+
+
+/**
+ * @author Gerolf Scherr
+ */
+public class EnhancedTextField extends JTextField {
```

Comment:
I see that we always set `selectAllOnFocus` to `true` which makes sense ebcause otherwise it acts as `JTextField`, right? I wonder if we really want this field to always "select all" when gaining the focus or just when opening the dialog, if we do that only when opening the dialog then we can replace this class with a call to `selectAll`..

Reply:
the select-all is the behaviour of TC (which I am quite used to), and I think the text field shouldn't behave differently when opening the dialog or jumping to the text field.

Reply:
ok, fair enough